### PR TITLE
move codec.deriveEnum and codec.deriveFixed to vulcan-generic

### LINKED ISSUE #241: Move `deriveEnum` and `deriveFixed` to generic module
These methods aren't straightforward to port to scala 3, and they feel more appropriate for the `generic` module anyway. As discussed in #239 we should first duplicate/alias them in the generic module and deprecate the core versions, then remove them from core in the next major release.

### PR DESCRIPTION
**Pull request by bplommer**
Resolves #241. I think we can release 1.2.0 after this.

Unfortunately this requires `vulcan-enumeratum` to depend on `vulcan-generic`. I don't see that as a major problem, but we could alternatively create a new `vulcan-generic-enum` module for deriveEnum

### CODE REVIEW

File: docs/src/main/mdoc/modules.md [hunk 1]
```diff
@@ -114,6 +115,56 @@ import shapeless.{:+:, CNil}
...
+  decode = {
+    case "apple"  => Right(Apple)
+    case "banana" => Right(Banana)
+    case "cherry" => Right(Cherry)
+    case other    => Left(AvroError(s"$other is not a Fruit"))
+  }
+)
+```
+
+`Codec.deriveFixed` can be used to partly derive [`Codec`][codec]s for fixed types. Annotations like `@AvroDoc` can be used to customize the derivation.
```
Comment (bplommer) [inline docs/src/main/mdoc/modules.md]:
```suggestion
`deriveFixed` can be used to partly derive [`Codec`][codec]s for fixed types. Annotations like `@AvroDoc` can be used to customize the derivation.
```

**Review by vlovgr — APPROVED** [review]
LGTM. 👍 

I was thinking whether we should name it `deriveEnumeration` now that we've renamed `Codec.enum` to `enumeration`? I don't mind either way.

**Comment by bplommer** [comment]
> I was thinking whether we should name it `deriveEnumeration` now that we've renamed `Codec.enum` to `enumeration`? I don't mind either way.

Feels a bit wordy to me, and `enum` is the official name in Avro.

I'll try drafting a release.
